src/network/plugins/base/base-network-plugin.ts

```diff
@@ -47,36 +50,48 @@ export default class BaseNetworkPlugin extends Events {
         throw `Invalid string PeedID [${peerId}]: ${e.message}`;
       }
     }
+
+
     try {
       let peer: Libp2pPeer = await this.network.libp2p.peerStore.get(peerId)
         .catch(e => null)
-      if(peer && peer.addresses.length) {
-        this.defaultLogger(`peer found local %p`, peerId)
+
+      if (peer && peer.addresses.length && this.hasValidTimestamp(peer)) {
+        this.defaultLogger(`peer found local %p`, peerId);
         return {
           id: peerId,
           multiaddrs: peer.addresses.map(addr => addr.multiaddr),
           protocols: []
         };
       }
-      this.defaultLogger(`peer not found local %p`, peerId)
+      this.defaultLogger(`peer not found local %p`, peerId);
       let routingPeer = await this.network.libp2p.peerRouting.findPeer(peerId);
 
+      peer = await this.network.libp2p.peerStore.get(peerId)
+        .catch(e => null);
+
       // There is a bug on libp2p 0.45.x
       // When a node dial another node, peer.addresses does not
       // save correctly on peerStore.
       // https://github.com/libp2p/js-libp2p/issues/1761
       //
       // We load addresses from peerRouting and patch the
       // peerStore
-      if(peer && !peer.addresses.length){
-        try{
+      if (peer) {
```

Comment:
Is this condition necessary?
This condition could be removed if patching a peer is possible even when it is not on peerstore.